aws_list_all/resource_filter.py

```diff
@@ -0,0 +1,285 @@
+        if listing.service == 'ec2' and listing.operation == 'DescribeRouteTables':
+            response['RouteTables'] = [
+                rt for rt in response['RouteTables'] if not any(x['Main'] for x in rt['Associations'])
+            ]
+
+class EC2NetworkAclsFilter:
+    def execute(self, listing, response):
+        # Filter default Network ACLs
+        if listing.service == 'ec2' and listing.operation == 'DescribeNetworkAcls':
+            response['NetworkAcls'] = [nacl for nacl in response['NetworkAcls'] if not nacl['IsDefault']]
+
+class EC2InternetGatewaysFilter:
+    def __init__(self, directory):
+        self.directory = directory
+
+    def execute(self, listing, response):
+        # Filter default Internet Gateways
+        if listing.service == 'ec2' and listing.operation == 'DescribeInternetGateways':
+            #describe_vpcs = run_raw_listing_operation(self.service, self.region, 'DescribeVpcs', self.profile)
+            vpcs_file = '{}_{}_{}_{}.json'.format(listing.service, 'DescribeVpcs', listing.region, listing.profile)
+            vpcs_file = self.directory + vpcs_file
+            vpcs_listing = Listing.from_json(json.load(open(vpcs_file, 'rb')))
+            describe_vpcs = vpcs_listing.response
+            vpcs = {v['VpcId']: v for v in describe_vpcs.get('Vpcs', [])}
+            internet_gateways = []
+            # print(self.response)
+            # print(response)
+            for ig in response['InternetGateways']:
+                attachments = ig.get('Attachments', [])
+                # more than one, it cannot be default.
+                if len(attachments) != 1:
+                    continue
+                vpc = attachments[0].get('VpcId')
+                if not vpcs.get(vpc, {}).get('IsDefault', False):
+                    internet_gateways.append(ig)
+            response['InternetGateways'] = internet_gateways
+
+class EC2FpgaImagesFilter:
+    def execute(self, listing, response):
+        # Filter Public images from ec2.fpga images
+        if listing.service == 'ec2' and listing.operation == 'DescribeFpgaImages':
+            response['FpgaImages'] = [image for image in response.get('FpgaImages', []) if not image.get('Public')]
+
+class WorkmailDeletedOrganizationsFilter:
+    def execute(self, listing, response):
+        # Remove deleted Organizations
+        if listing.service == 'workmail' and listing.operation == 'ListOrganizations':
+            response['OrganizationSummaries'] = [
+                s for s in response.get('OrganizationSummaries', []) if not s.get('State') == 'Deleted'
+            ]
+
+class ElasticacheSubnetGroupsFilter:
+    def execute(self, listing, response):
+        if listing.service == 'elasticache' and listing.operation == 'DescribeCacheSubnetGroups':
+            response['CacheSubnetGroups'] = [
+                g for g in response.get('CacheSubnetGroups', []) if g.get('CacheSubnetGroupName') != 'default'
+            ]
```

Comment:
This branch probably needs a rebase to the current master since I merged the other PR